operator/validator/controller.go

```diff
@@ -175,9 +176,9 @@ type controller struct {
 	historySyncBatchSize int
 	messageValidator     validation.MessageValidator
 
-	// nonCommittees is a cache of initialized committeeObserver instances
-	committeesObservers      *ttlcache.Cache[spectypes.MessageID, *committeeObserver]
-	committeesObserversMutex sync.Mutex
+	// committeeObservers is a cache of initialized committeeObserver instances
+	committeesObservers      *hashmap.Map[spectypes.MessageID, *committeeObserver] // todo: need to evict?
```

Comment:
we don't have to, it's an optimization to stay lean by evicting:
- removed validators
- proposal duties

if you don't delete them they accumulate over time, however we can try without it and revert if needed

Reply:
I agree we should eventually evict but I felt ttl cache might not be the best here. maybe we could somehow remove it on actual events in the future.